### PR TITLE
Add field to keep human-readable information of authentication policy.

### PR DESCRIPTION
**Pull request by diemtvu**
- Add 'description' to authentication policy for human readable note.
- Add policy_name to authentication filter config to show the corresponding policy that pilot used to config filter (for debugging purpose).

### CODE REVIEW

File: envoy/config/filter/http/authn/v2alpha1/config.proto [hunk 1]
```diff
@@ -33,4 +33,7 @@ message FilterConfig {
   // This information is added by pilot when construct and add Jwt and
   // authN filters.
   map<string, string> jwt_output_payload_locations = 2;
+
+  // Name of the policy above in the istio config storage. This is for debugging purpose.
```
Comment (incfly) [inline envoy/config/filter/http/authn/v2alpha1/config.proto]: Why we need this? I think policy itself already has the name?

Comment (diemtvu) [reply envoy/config/filter/http/authn/v2alpha1/config.proto]: CRD has name, but the config we send from pilot to envoy contain the proto only.

Comment (incfly) [reply envoy/config/filter/http/authn/v2alpha1/config.proto]: (up to you) but do we need this? We add `description` above and that will be propagated from pilot to sidecar for free. This field I assume you need to change pilot code to copy from CRD name?

**Comment by diemtvu** [comment]
Yeah, I plan to populate policy name in pilot so it's easier to check. However, it might make the code a bit ugly.

Adding extra text field to the user facing API may not be a good approach neither.

 Close this PR for now until we needed.

**Pull request closed without merging**